dlc/src/lib.rs

```diff
@@ -0,0 +1,703 @@
+        &fund_tx_in,
+        &local_params.final_address,
+        &remote_params.final_address,
+        outcomes,
+        lock_time,
+    );
+
+    let local_refund_output = TxOut {
+        value: local_params.collateral,
+        script_pubkey: local_params.final_address.script_pubkey(),
+    };
+
+    let remote_refund_ouput = TxOut {
+        value: remote_params.collateral,
+        script_pubkey: remote_params.final_address.script_pubkey(),
+    };
+
+    let refund_tx = create_refund_transaction(
+        local_refund_output,
+        remote_refund_ouput,
+        fund_tx_in,
+        refund_lock_time,
+    );
+
+    Ok(DlcTransactions {
+        fund: fund_tx,
+        cets,
+        refund: refund_tx,
+    })
+}
+
+/// Create a contract execution transaction
+pub fn create_cet(
+    local_output: TxOut,
+    remote_output: TxOut,
+    fund_tx_in: &TxIn,
+    lock_time: u32,
+) -> Transaction {
+    let output = [local_output, remote_output]
+        .iter()
+        .filter(|o| o.value >= DUST_LIMIT)
+        .cloned()
+        .collect();
+
+    Transaction {
+        version: TX_VERSION,
+        lock_time,
+        input: vec![fund_tx_in.clone()],
+        output,
+    }
+}
+
+/// Create a set of contract execution transaction for each provided outcome
+pub fn create_cets(
+    fund_tx_input: &TxIn,
+    local_final_address: &Address,
+    remote_final_address: &Address,
+    outcomes: &[Outcome],
+    lock_time: u32,
+) -> Vec<Transaction> {
+    let mut txs: Vec<Transaction> = Vec::new();
+    for i in 0..outcomes.len() {
+        let local_output = TxOut {
+            value: outcomes[i].local,
+            script_pubkey: local_final_address.script_pubkey(),
+        };
+        let remote_output = TxOut {
+            value: outcomes[i].remote,
+            script_pubkey: remote_final_address.script_pubkey(),
+        };
+        let tx = create_cet(local_output, remote_output, fund_tx_input, lock_time);
+        txs.push(tx);
+    }
+
+    txs
+}
+
+/// Create a funding transaction
+pub fn create_funding_transaction(
+    local_fund_pubkey: &PublicKey,
+    remote_fund_pubkey: &PublicKey,
+    output_amount: u64,
+    local_inputs: &[TxIn],
+    remote_inputs: &[TxIn],
+    local_change_output: TxOut,
+    remote_change_output: TxOut,
+    lock_time: u32,
+) -> Transaction {
+    let script = make_funding_redeemscript(&local_fund_pubkey, &remote_fund_pubkey);
+
+    let output: Vec<TxOut> = {
+        let outs = vec![
+            TxOut {
+                value: output_amount,
+                script_pubkey: script.to_v0_p2wsh(),
+            },
+            local_change_output,
+            remote_change_output,
+        ];
+        outs.into_iter().filter(|o| o.value >= DUST_LIMIT).collect()
+    };
+
+    let input = [&local_inputs[..], &remote_inputs[..]].concat();
+
+    let funding_transaction = Transaction {
+        version: TX_VERSION,
+        lock_time: lock_time,
+        input,
+        output,
+    };
```

Comment:
As I mentioned [here](https://github.com/discreetlogcontracts/dlcspecs/pull/57#discussion_r484186590) I'm not sure we should enforce change output for v1. But if you convince me for the specs I can easily make the change here :)

Reply:
Depends. How easy it should be to break a V0.1 DLC :) ?

Reply:
I still haven't been convinced that having the change output provides more than convenience, if it is a security issue I am happy to enforce it (and deal with the extra burden), but otherwise I fund it more inconvenient because:
* It complicates coin selection. If it finds a UTXO that satisfies the collateral amount, I actually need to re-run with a different amount until I find something that actually have a change
* It increases transaction size and so fee
* Increases UTXO set of user